### PR TITLE
Add customer-supplied encryption keys to Storage

### LINKED ISSUE #722: Support custom encryption keys for storage?
Can we support custom encryption keys for files uploaded?

Spec is here: https://cloud.google.com/compute/docs/disks/customer-supplied-encryption#top_of_page

### PR DESCRIPTION
**Pull request by quartzmo**
This implementation does not address a known issue in which the
service returns 411 when request sent with httpclient.

[closes #722]

### CODE REVIEW

File: lib/gcloud/storage/bucket.rb [hunk 1]
```diff
@@ -433,9 +446,9 @@ def files prefix: nil, delimiter: nil, token: nil, max: nil, versions: nil
       #   file = bucket.file "path/to/my-file.ext"
       #   puts file.name
       #
-      def file path, generation: nil
+      def file path, generation: nil, key: nil, key_sha256: nil
```
Comment (jgeewax) [inline lib/gcloud/storage/bucket.rb]: encryption_key * ?

Comment (quartzmo) [inline lib/gcloud/storage/bucket.rb]: @blowmage? (My personal preference is for long, explicit names that hold closely to the underlying API...)

Comment (blowmage) [inline lib/gcloud/storage/bucket.rb]: My preference is for `key`.

**Comment by quartzmo** [comment]
@blowmage Option names updated, ready for review.

**Pull request merged**